Tone/signal/Scale.ts

```diff
@@ -0,0 +1,104 @@
+import { ToneAudioNodeOptions } from "Tone/core/context/ToneAudioNode";
+import { optionsFromArguments } from "Tone/core/util/Defaults";
+import { Add } from "./Add";
+import { Multiply } from "./Multiply";
+import { Signal } from "./Signal";
+import { SignalOperator } from "./SignalOperator";
+
+export interface ScaleOptions<Type> extends ToneAudioNodeOptions {
+	value: Type;
+	outputMin: number;
+	outputMax: number;
+}
+
+/**
+ *  @class  Performs a linear scaling on an input signal.
+ *          Scales a NormalRange input to between
+ *          outputMin and outputMax.
+ *
+ *  @constructor
+ *  @extends {SignalOperator}
+ *  @param outputMin The output value when the input is 0.
+ *  @param outputMax The output value when the input is 1.
+ *  @example
+ * var scale = new Scale(50, 100);
+ * var signal = new Signal(0.5).connect(scale);
+ * //the output of scale equals 75
+ */
+export class Scale extends SignalOperator<ScaleOptions<number>> {
+
+	readonly name: string = "Scale";
+
+	input = new Multiply({
+		context: this.context,
+		value: 1,
+	});
+
+	output = new Add({
+		context: this.context,
+		value: 0,
+	});
+
+	value: number = 0;
+
+	private _outputMin!: number;
+
+	private _outputMax!: number;
+
+	constructor(options?: Partial<ScaleOptions<number>>);
+	// tslint:disable-next-line: unified-signatures
+	constructor(value?: number);
+	constructor() {
+		super(Object.assign(optionsFromArguments(Scale.getDefaults(), arguments, ["value"])));
+
+		const options = optionsFromArguments(Scale.getDefaults(), arguments, ["outputMin", "outputMax"]);
+		this._outputMin = options.outputMin;
+		this._outputMax = options.outputMax;
+
+		this.connect(this.output);
```

Comment:
i think there's an issue with this line. it should probably be `this.input.connect(this.output)`

Reply:
I tried that, still getting the tests to fail but the error seems more meaningful now, I'm getting
```AssertionError: expected -9 to be close to 8 +/- 0.01 ```
```AssertionError: expected 10.5 to be close to 15 +/- 0.01 ```

Reply:
found the issue, I was using the incorrect property inside _setRange